packages/circuit-ui/components/ComparisonTable/components/PlanTable/PlanTable.tsx

```diff
@@ -0,0 +1,200 @@
+  /**
+   * A descriptive caption for the table.
+   */
+  caption: string;
+  /**
+   * The label for the show all features button.
+   */
+  showAllFeaturesLabel: string;
+  /**
+   * A list of sections with features to display.
+   */
+  sections: FeatureSection[];
+  /**
+   * A list of the items to compare.
+   */
+  headers: TableHeaderProps[];
+  /**
+   * An array of the two positions of the currently selected plans.
+   * the first index is the first plan, the second index is the second plan.
+   */
+  activePlans: number[];
+}
+
+export const PlanTable = forwardRef<HTMLTableElement, PlanTableProps>(
+  (
+    { caption, showAllFeaturesLabel, headers, sections, activePlans, ...props },
+    ref,
+  ) => {
+    const tableRef = useRef<HTMLTableElement>(null);
+    const isMobile = useMedia('(max-width: 767px)');
+    const isTablet = useMedia('(max-width: 767px) and (min-width: 480px)');
+    const [isCollapsed, setIsCollapsed] = useState(
+      sections.reduce(
+        (totalRows, section) => totalRows + section.features.length + 1,
+        0,
+      ) > 8,
+    );
+    const theadRef = useRef<HTMLTableSectionElement>(null);
+    const [headerHeight, setHeaderHeight] = useState(0);
+    const rowsToDisplay = isCollapsed ? getFirstEightRows(sections) : sections;
+
+    const updateHeaderHeight = useCallback(() => {
+      throttle(() => {
+        setHeaderHeight(theadRef.current?.getBoundingClientRect().height ?? 0);
+      }, 500)();
+    }, []);
+
+    // biome-ignore lint/correctness/useExhaustiveDependencies: we want to update this value when the isCollapsed prop changes
+    useEffect(() => {
+      updateHeaderHeight();
+    }, [updateHeaderHeight, isCollapsed, isMobile]);
+
+    useEffect(() => {
+      window.addEventListener('resize', updateHeaderHeight);
+      return () => {
+        window.removeEventListener('resize', updateHeaderHeight);
+      };
+    }, [updateHeaderHeight]);
+
+    const showFeatures = useCallback(() => {
+      setIsCollapsed(false);
+    }, []);
+
+    useEffect(() => {
+      if (!isCollapsed) {
+        // set focus to the first row of the added rows when the table is expanded
+        tableRef.current
+          ?.querySelectorAll('tr')[9]
+          ?.focus({ preventScroll: true });
+      }
+    }, [isCollapsed]);
+
+    const SectionTitleElement = isMobile ? Compact : Body;
+
+    return (
+      <div className={classes.wrapper}>
+        <table
+          className={classes.base}
+          ref={applyMultipleRefs(ref, tableRef)}
+          {...props}
+        >
+          <caption id="caption" className={utilClasses.hideVisually}>
+            {caption}
+          </caption>
+          <colgroup>
+            <col />
+            {(isMobile ? activePlans : headers).map((_, index) => (
+              <col key={index} />
+            ))}
+          </colgroup>
+          <thead ref={theadRef}>
+            <tr>
+              <th id="features" aria-hidden={true} />
```

Comment:
This id is quite generic and could clash with other ids on the page. Either prefix it with `cui-` or use the `useId` hook to reduce the risk of collision.